setup.js

```diff
@@ -1,341 +1,352 @@
+const fetch = require("node-fetch");
+const inquirer = require("inquirer");
+const Locale = require("./locale");
+global.version = "1.4.2";
+global.delay = (ms) => new Promise((res) => setTimeout(res, ms));
+global.cacheObj = {};
+global.clearLastLine = () => {
+	process.stdout.moveCursor(0, -1);
+	process.stdout.clearLine(1);
+};
+global.getCompiled = () => {
+	let path = require.main.path;
+	let splitPath = path.split("\\");
+	splitPath.pop();
+	let name = splitPath.pop();
+	if (name === "snapshot") return true;
+	return false;
+};
+global.writeCache = async () => {
+	let path = getConfigPath();
+	path += "cache.json";
+	await fs.writeFileSync(path, JSON.stringify(cacheObj, null, "\t"));
+};
+global.configObj = {};
+let defaultConfig = {
+	updateDetected: false,
+	unfinishedDownloads: [],
+};
+
+const SETUP_PREFIX = `\x1b[34m[${Locale.get("SETUP")}]\x1b[0m`;
+const ERROR_PREFIX = `\x1b[31m[${Locale.get("ERROR")}]\x1b[0m`;
+const INFO_PREFIX = `\x1b[36m[${Locale.get("INFO")}]\x1b[0m`;
+
+module.exports = async () => {
+	console.log(SETUP_PREFIX, Locale.get("CHECKING_PYTHON_VERSION"));
+	let py_ver;
+	try {
+		result = cp.execSync("py --version", { stdio: "pipe" }).toString();
+		py_ver = result.slice(7, result.length - 2);
+	} catch (e) {
+		if (e.toString() === "Error: Command failed: py --version") {
+			console.log(
+				ERROR_PREFIX,
+				"Python not installed! First install it in order to use this tool! (Remember to add it to the path)"
+			);
+			await delay(5000);
+			process.exit(1);
+		}
+	}
+
+	let installedVersion = py_ver.split(".").map((i) => parseInt(i));
+	let compatible = true;
+	if (installedVersion[0] < 3 && installedVersion[1] < 9) compatible = false;
+
+	if (!compatible) {
+		console.log(
+			ERROR_PREFIX,
+			`Incompatible Python detected (${py_ver}), you need Python 3.9+ to run this program! Exiting in 5 seconds...`
+		);
+		await delay(5000);
+		process.exit(1);
+	}
+
+	console.log(
+		SETUP_PREFIX,
+		`${Locale.get(
+			"COMPATIBLE_PYTHON_VERSION_DETECTED"
+		)} (${py_ver})! ${Locale.get("PROCEEDING")}...`
+	);
+
+	const configPath = getConfigPath();
+
+	if (!(await fs.existsSync(configPath + "config.json"))) {
+		console.log(
+			SETUP_PREFIX,
+			Locale.get("CONFIG_FILE_NOT_FOUND_CREATING_IT")
+		);
+		createConfig();
+		await updateLegendary();
+	} else {
+		console.log(
+			SETUP_PREFIX,
+			`${Locale.get("CONFIG_FILE_FOUND")} ${Locale.get("PROCEEDING")}...`
+		);
+		await checkConfig();
+		checkForUpdateLegendary();
+	}
+
+	console.log(SETUP_PREFIX, Locale.get("CHECKING_SESSION") + "...");
+	await loginOntoLegendary();
+
+	console.log(SETUP_PREFIX, Locale.get("LOADING_CACHE") + "...");
+	await loadCache();
+	console.log(SETUP_PREFIX, Locale.get("FINISHED_LOADING_CACHE"));
+
+	console.log(INFO_PREFIX, Locale.get("CHECKING_FOR_UPDATES") + "...");
+	const lastVersion = await isUpdateAvailable();
+	if (lastVersion)
+		console.log(
+			INFO_PREFIX,
+			Locale.get("UPDATE_DETECTED") + ` (v${lastVersion})`

```

Comment:
```suggestion
			Locale.get("UPDATE_AVAILABLE") + ` (v${lastVersion})`
```
UPDATE_DETECTED doesn't exist, it should be UPDATE_AVAILABLE instead.